pylint/reporters/json_reporter.py

```diff
@@ -24,24 +27,47 @@ class JSONReporter(BaseReporter):
 
     def display_messages(self, layout: Section | None) -> None:
         """Launch layouts display."""
-        json_dumpable = [
-            {
-                "type": msg.category,
-                "module": msg.module,
-                "obj": msg.obj,
-                "line": msg.line,
-                "column": msg.column,
-                "endLine": msg.end_line,
-                "endColumn": msg.end_column,
-                "path": msg.path,
-                "symbol": msg.symbol,
-                "message": msg.msg or "",
-                "message-id": msg.msg_id,
-            }
-            for msg in self.messages
-        ]
+        json_dumpable = [self.serialize(message) for message in self.messages]
         print(json.dumps(json_dumpable, indent=4), file=self.out)
 
+    @staticmethod
+    def serialize(message: Message) -> dict[str, int | str | None]:
+        # TODO (3.0) add abs-path and confidence
+        return {
+            "type": message.category,
+            "module": message.module,
+            "obj": message.obj,
+            "line": message.line,
+            "column": message.column,
+            "endLine": message.end_line,
+            "endColumn": message.end_column,
+            "path": message.path,
+            "symbol": message.symbol,
+            "message": message.msg or "",
+            "message-id": message.msg_id,
+        }
+
+    @staticmethod
+    def deserialize(message_as_json: dict) -> Message:
+        return Message(
+            msg_id=message_as_json["message-id"],
+            symbol=message_as_json["symbol"],
+            msg=message_as_json["message"],
+            location=MessageLocationTuple(
+                # TODO (3.0) abs-path is not available in json export
```

Comment:
```suggestion
                # TODO: 3.0: Add abs-path and confidence or a new JSONReporter
```

Reply:
I think we should definitely have two JSON reporters in 2.x series,  and keep only the new one in 3.0.